micrometer-core/src/main/java/io/micrometer/core/instrument/composite/AbstractCompositeMeter.java

```diff
@@ -49,9 +49,11 @@ final Iterable<T> getChildren() {
     }
 
     T firstChild() {
-        final Iterator<T> i = children.values().iterator();
-        if (i.hasNext())
-            return i.next();
+        for (T next : children.values()) {
+            if (!(next instanceof NoopMeter)) {
+                return next;
+            }
+        }
 
         // There are no child meters. Return a lazily instantiated no-op meter.
         final T noopMeter = this.noopMeter;
```

Comment:
If we are at this point, the `children` collection was either empty or contains only noop meters. In the second case we could return it the first value (like we did before this change) but I'm not sure it worth the hassle.